forward/utils.go

```diff
@@ -17,9 +17,27 @@ func cloneURL(i *url.URL) *url.URL {
 // copyHeaders copies http headers from source to destination.  It does not
 // overide, but adds multiple headers
 func copyHeaders(dst, src http.Header) {
+	var extraHopByHopHeaders []string
 	for k, vv := range src {
-		for _, v := range vv {
-			dst.Add(k, v)
+		switch k {
```

Comment:
I looked it up, and most of the compilers will turn the switch into an array, especially if there are many cases in the switch. Apparently in C, the -O2 generates faster code for a switch, but -O3 generates faster code for an array, so I would go for the cleanest implementation.